docs/eventing/experimental-features.md

```diff
@@ -0,0 +1,60 @@
+---
+title: "Experimental features"
+weight: 99
+type: "docs"
+---
+
+# Experimental features
+
+In Knative Eventing we want to keep the innovation alive, experimenting and delivering new features without affecting the stability of the project.
+
+In order to achieve that goal, we have a process to include new features.
+This allows users like you to try out new features and provide feedback back to the project.
+
+This document explains how to enable experimental features and which ones are available today.
+
+For more details about the process, the feature phases, quality requirements and guarantees, check out the [Experimental features process documentation](https://github.com/knative/eventing/blob/main/docs/experimental-features.md).
+
+## Before you begin
+
+You must have a Knative cluster running with the Eventing component installed. [Learn more](../install/)
+
+## Experimental features configuration
+
+When installing Eventing, the `config-experimental-features` ConfigMap is added to your cluster in the `knative-eventing` namespace.
+In order to enable a feature, you just need to add it to the config map and set its value to `"true"`.
+For example, to enable `new-cool-feature`:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-experimental-features
```

Comment:
ditto that this is really a comment on the implementation not the doc, but making it here because this is where Im seeing it -- could this be `config-features` to align with the same thing in knative-serving namespace? (I'd say we should rename the serving one, but that would pose a backwards-compat challenge now).

Reply:
UX feedback 
I think the word “experimental” should not be in any configuration to indicate the level of stability or to indicate that the configuration/feature might get removed

a better approach perhaps would be using docs to indicate this
For example a document comment in the yaml
A message on the command line like when using deprecated features in kubectl, they do this in Kubernetes know and even with colors text yellow to indicate warning
In the action website docs in both the section explaining how to use the feature and in the Go generate re fence html 

the benefit would be that as the user upgrades from old version to new version, and it happens that the feature is stable and not longer experimental 
There is not much work for the administrators to go and update the config in many places or coordinate with others to change the config in many places. Instead he can announce/doc that a new version of Knative is used and that feature X is not longer experimental 

for Knative to move it out from experimental we don’t modify tests or code. 

If the feature is removed from experimental then is a mute point the experience remains the same for that case as oppose feature remaining they have better experience

Reply:
So this CM is an entrypoint for people that wants to try out experimental features, and when a feature is stable, the flag won't be here anymore. The name experimental is just to keep it consistent with the _Experimental features process_.

The CM gets commented too with details about each specific flag, and then this page will host the documentation for each flag.

> For example a document comment in the yaml
A message on the command line like when using deprecated features in kubectl, they do this in Kubernetes know and even with colors text yellow to indicate warning
In the actual website docs in both the section explaining how to use the feature and in the Go generate rerefence html

For experimental features adding new fields to CRDs, we don't add them to the schema: https://github.com/knative/eventing/blob/main/docs/experimental-features.md#strategies-to-add-new-apis-for-experimental-features

Reply:
> So this CM is an entrypoint for people that wants to try out experimental features, and when a feature is stable, the flag won't be here anymore.

Sure but like @julz and @csantanapr have pointed out someone coming from serving to enable some functionality will now have a _different_ approach will easily be confused and tripped up.